lib/matplotlib/bezier.py

```diff
@@ -68,6 +68,54 @@ def get_normal_points(cx, cy, cos_t, sin_t, length):
     return x1, y1, x2, y2
 
 
+# deprecated routines (moved to `.Path`)
+
+
+@cbook.deprecated("3.2")
+def inside_circle(cx, cy, r):
+    """
+    Return a function that checks whether a point is in a circle with center
+    (*cx*, *cy*) and radius *r*.
+
+    The returned function has the signature::
+
+        f(xy: Tuple[float, float]) -> bool
+    """
+    r2 = r ** 2
+
+    def _f(xy):
+        x, y = xy
+        return (x - cx) ** 2 + (y - cy) ** 2 < r2
+    return _f
+
+
+@cbook.deprecated("3.2", alternative="Path.make_compound_path()")
+def split_path_inout(path, inside, tolerance=0.01, reorder_inout=False):
```

Comment:
You can just reimport this function at the top of the module (`from .path import split_path_inout` with a note that this needs to remain available in this module for API stability reasons) and then not bother with the deprecation (if the implementation stays around anywhere in the library, the disruption of removing it isn't really worth it).

Reply:
The idea behind this pull request is to remove `from .path import ...` from the top of `bezier.py`. This is because I implemented all the functions to do e.g. MarkerStyle renormalization (and other things), but they are all methods of `Path`, and they need access to `BezierSegment`. Currently, I cannot `from .bezier import ...` in `path.py` because `bezier.py` already imports `from .path ...`. So reimporting this function at the top of the module would defeat the purpose.

Reply:
Ah.  OK, but then at the *end* of path.py you can do `bezier.split_path_inout = Path.split_path_inout` (with a comment re: conserving old API)?  This way people doing `from bezier import split_path_inout` won't be affected?

Reply:
Maybe I'm missing something, but as in other comment, it seems like this patch would only work if it was applied at the package level? For example in `lib/matplotlib/__init__.py` I could add 

```
import matplotlib.patches
import matplotlib.path
import matplotlib.bezier
bezier.split_path_inout = Path.split_path_inout
bezier.inside_circle = patches._inside_circle
```

but I don't see how that is better than my current approach, of just keeping the functions in `bezier.py` and calling their new implementations. My approach also doesn't have any repeated code, and doesn't affect anybody doing `from bezier import split_path_inout`. 

If your argument is that we shouldn't deprecate these functions at all, I am happy to remove the `@deprecated` statements, since the *real* point of this PR was to remove the `from .path import Path` statement from `bezier.py`.

Reply:
it is better to keep these sort of details contained inside of the modules that affect them.  Having `path.py` monkey-patch `bezier` is less bad than doing the monkey patching in the top level `__init__.py`

It looks like you are moving all of the helper methods from `bezier.py` -> `path.py` then it is OK to keep a `.path` import in `bezier.py`?

Reply:
@tacaswell, I agree. That's why I didn't do it in the first place. 

Maybe I'm missing something, but the proposed monkey patch doesn't solve the stated problem (of not changing the public API)? 

I think I haven't been clear enough in this PR: the point is to remove the `.path` import from `bezier.py`. Why? Because many things that you might want to compute about Paths (bounding box, area, center of mass, see e.g. #16832, #16859) need to call functions of `bezier.py`. 

Conversely, the current reason for the existence of the import is that a bunch of functions that have no business being in `bezier.py` in the first place.

So one solution is to move the `.path` import into the functions.

Another (this one), is to move the functions where they ought to go and have `bezier.whatever` just pass thru to the new version of the function. (And optionally deprecate the old versions).

Compared to the pass-thru approach, the monkey patch approach proposed by @anntzer literally does not work.

With my current approach:

```
from matplotlib import bezier
bezier.split_path_inout
```

Works.

With @anntzer's monkey patch approach as I understand it, you need to do

```
from matplotlib import bezier
bezier.split_path_inout # error
from matplotlib import path
bezier.split_path_inout # works
```

Sorry if reply sounds terse, I'm on mobile.

Reply:
But if you have moved all of the useful functions to `path.py` it no longer _has_ to import `bezier` so it is no longer circular.  Are there more functions than these that you want to move?

We could also pull the common code out to a new module that both `path.py` and `bezier.py` import from.

As a general rule, we try to avoid both API changes and code-churn for the sake of aesthetics changes.  The question I have is: Is this is the _minimal_ API and code change that we can do to get to the desired outcome?

Reply:
Ah, I see where you're coming from now, it was just me not doing a good job explaining why this PR exists in the first place. 

This is the minimum API change so that I can do #16832, #16859 (and more), which are not-just-aesthetic. Basically the confusion seems to be that since those new functions aren't in this PR, the PR doesn't seem justified. 

In short, I moved all the (currently) useful functions to `path.py`. There not more functions that I want to move, but there are a half-dozen functions that I have written that depend on `.path` being able to `from bezier import BezierSegment` (again, that class is empty in this PR, but that's only because I wanted to split the PRs into manageable chunks to make review easier.

Reply:
Maybe it makes sense for me to make a "BIG" PR of my "equal area markers" branch (which I currently have split into #16812, #16832, #16859, (and one more that I haven't pushed up yet).

That way we can discuss the top-level planning of how to go about the equal area markers thing? Given my current strategy (writing code to compute the area of an arbitrary path, which turns out to be quite simple, see #16859), this PR is a necessary first step.